omegaconf/base.py

```diff
@@ -795,6 +795,12 @@ class SCMode(Enum):
     INSTANTIATE = 3  # Create a dataclass or attrs class instance
 
 
+class ListMergeMode(Enum):
+    OVERRIDE = 1  # content from newer list gets taken (default)
+    EXTEND = 2  # lists get extended
+    EXTEND_IGNORE_DUPLICATES = 3  # only new (unique) elements get extended
```

Comment:
I know we have went though several iteration of bikeshedding here already, but any thoughts on using append instead of extend?
Also I think for the default REPALCE is more appropriate than override.
```suggestion
class ListMergeMode(Enum):
    REPALCE = 1  # Replaces the target list with the new one (default)
    APPEND= 2   # Appends the new list to the target list
    APPEND_UNIQUE_VALUE = 3  # Appends items that do not already exist in the target list
```

Reply:
I think `APPEND` is confusing due to the way that python's builtin `list.append` method works:
```python
>>> lst = [1,2,3]; lst.extend([4,5,6]); print(lst)  # expected
[1, 2, 3, 4, 5, 6]
>>> lst = [1,2,3]; lst.append([4,5,6]); print(lst)  # unexpected
[1, 2, 3, [4, 5, 6]]
```

Reply:
> Also I think for the default REPALCE is more appropriate than override.

👍🏼

Reply:
I like `REPLACE` and `APPEND_UNIQUE_VALUE` a lot. But I agree to Jasha, that we should name it after the python functions, so that `extend` is __extending__ the list. For `APPEND_UNIQUE_VALUE` I like the way, that we don't extend lists anymore, but rather looking at the actual values and if they are unique in the target list. @omry ,what do you think?

Reply:
Okay, from the perspective of the target list:
```python
# replace target list
REPLACE
# Extend target list with new list
EXTEND
# Extend target list with items not already present in it.
EXTEND_UNIQUE 
```